### DISCUSSION TITLE
How to disable/remove default podDisruptionBudget rules for kafka and zookeeper?

### DISCUSSION THREAD

**Discussion by bharat15-ml** [strimzi/strimzi-kafka-operator#8136] [Q&A]
Hi All,

Do we have any idea/thoughts on how to remove default podDisruptionBudget rules for strimzi component?
Due to pdb rules, unable to upgrade kubernetes cluster.

Thanks!!

**Comment by scholzj** [comment]
You cannot remove or disable them. Why can't you upgrade your Kubernetes cluster because of them? Without them, your cluster might be unavailable during the upgrade.

**Comment by bharat15-ml** [comment]
Thanks Scholzj, with valid point, completely agree with it. 
What we have observed sometimes, If we have pdb's objects configured that are unable to allow any additional disruptions, node upgrades might fail to upgrade the control plane version. We are able to deal with it by following some manual steps. Thanks!!

**Reply by scholzj** [reply]
Well, the PDB is there to make sure no more than a given number of pods is offline at a particular point in time due to voluntary disruptions such as node draining. What you describe sounds like not all your pods are ready. You can increase the number of pods that are allowed to be not ready at the same time if you want. But it would _weaken_ your availability.
